### PR TITLE
QA-4728 - Added the BW/Users/Details endpoint

### PR DESCRIPTION
**Pull request by bw-lee**
Added a missing endpoint - `Brandwatch/users/details` 

***Rubocop***
153 files inspected, no offenses detected

### CODE REVIEW

File: lib/bwapi/client/brandwatch/users.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
+module BWAPI
+  class Client
+    module Brandwatch
+      # users_details module for brandwatch/users/details endpoint
+      module Users
+        # Returns the specified users details
+        #
+        # @param opts [Hash] username
+        # @return [Array] Returns the user DTO
```
Comment (jonathanchrisp) [inline lib/bwapi/client/brandwatch/users.rb]: Does it return an array here? To me that would be odd but just double checking

File: lib/bwapi/client/brandwatch/users.rb [hunk 2]
```diff
@@ -0,0 +1,16 @@
+module BWAPI
+  class Client
+    module Brandwatch
+      # users_details module for brandwatch/users/details endpoint
+      module Users
+        # Returns the specified users details
+        #
+        # @param opts [Hash] username
```
Comment (jonathanchrisp) [inline lib/bwapi/client/brandwatch/users.rb]: Can we add the params below?

Comment (jonathanchrisp) [inline lib/bwapi/client/brandwatch/users.rb]: e.g `# @option opts [Integer] id Id of the user`

File: lib/bwapi/client/brandwatch/users.rb (hunk 1 above)
Comment (jonathanchrisp) [inline lib/bwapi/client/brandwatch/users.rb]: `Array` should be `Hash`

File: lib/bwapi/client/brandwatch/users.rb [hunk 3]
```diff
@@ -0,0 +1,16 @@
+module BWAPI
+  class Client
+    module Brandwatch
+      # users_details module for brandwatch/users/details endpoint
+      module Users
+        # Returns the specified users details
+        #
+        # @option opts [String] username of the user
```
Comment (jonathanchrisp) [inline lib/bwapi/client/brandwatch/users.rb]: Ah sorry this is slightly wrong, should be like:

```
# Gets all author professions
#
# @param opts [Hash] options hash of parameters
# @option opts [Integer] page Page of results to retrieve
# @option opts [Integer] pageSize Results per page of results
# @return [Hash] All author professions
```

Comment (jonathanchrisp) [inline lib/bwapi/client/brandwatch/users.rb]: Sorry

**Comment by bw-lee** [comment]
@benSlaughter  are you happy to merge?  I have the API tests ready to go, but waiting for the version no. first.

**Pull request merged**